tck/src/main/java/org/eclipse/microprofile/config/tck/ConfigPropertiesTest.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2016-2017 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.config.tck;
+
+import javax.inject.Inject;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.testng.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+/**
+ * Verify the support {@code ConfigProperties}.
+ *
+ * @author <a href="mailto:emijiang6@googlemail.com">Emily Jiang</a>
+ */
+public class ConfigPropertiesTest extends Arquillian {
+
+    private @Inject ConfigPropertyBeanOne beanOne;
+    private @Inject ConfigPropertyBeanTwo beanTwo;
+
+    @Deployment
+    public static WebArchive deploy() {
+        JavaArchive testJar = ShrinkWrap
+                .create(JavaArchive.class, "ConfigPropertiesTest.jar")
+                .addClasses(ConfigPropertiesTest.class, ConfigPropertyBeanOne.class, ConfigPropertyBeanTwo.class, Location.class)
+                .addAsManifestResource(
+                    new StringAsset(
+                        "customer.name=Bob\n" +
+                        "customer.age=24\n" +
+                        "customer.location=2 Hook Road, Winchester, Hampshire, SO21 2JN, UK\n" +
```

Comment:
I can see that this style would use the `Location` constructor which then splits the value to the fields. Naturally it makes me wonder if one also could have skipped the constructor and used nested beans when then would be defined like

```
customer.location.road=2 Hook Road
...
```

Reply:
yes, you can. It is not the point for the tck. I want to demonstrate you can inject other value types.